### PR TITLE
feat: create new command for the react template

### PR DESCRIPTION
**Pull request by AayushSaini101**
Resolves: #337 

* Create a new command to create a sample react template 
* Reference template format for react engine: https://www.asyncapi.com/docs/tools/generator/template-development.
* Added test cases

### CODE REVIEW

File: assets/create-template/templates/default/asyncapi.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+asyncapi: 2.6.0
```
Comment (Amzani) [inline assets/create-template/templates/default/asyncapi.yaml:1]: @AayushSaini101 I suggest to target version 3 instead, as we want people to adopt it.

File: assets/create-template/templates/default/package.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+    "name": "myTemplate",
+    "generator": {
+        "renderer": "react",
+        "supportedProtocols": [ ]
+    },
+    "dependencies": {
+        "@asyncapi/generator-react-sdk": "^0.2.25"
```
Comment (Amzani) [inline assets/create-template/templates/default/package.json:8]: I recommend to pick last version of generator-react-sdk 
https://github.com/asyncapi/generator-react-sdk/releases/tag/v1.0.20

File: assets/create-template/templates/default/template/package.json.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (Amzani) [inline assets/create-template/templates/default/template/package.json.js:1]: I'm not sure about the naming of this file and its purpose

File: src/core/flags/new/template.flags.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { Flags } from '@oclif/core';
+
+export const templateFlags = () => {
+  return {
```
Comment (Amzani) [inline src/core/flags/new/template.flags.ts:4]: I suggest to add also `renderer` flag as some people may use `nunjucks`, see: https://www.asyncapi.com/docs/tools/generator/template-development

**Review by Amzani — CHANGES_REQUESTED** [review]

File: src/core/flags/new/template.flags.ts (hunk 4 above)
Comment (AayushSaini101) [reply src/core/flags/new/template.flags.ts:4]: Done @Amzani

File: assets/create-template/templates/default/package.json (hunk 2 above)
Comment (AayushSaini101) [reply assets/create-template/templates/default/package.json]: Done @Amzani

File: assets/create-template/templates/default/template/package.json.js (hunk 3 above)
Comment (AayushSaini101) [reply assets/create-template/templates/default/template/package.json.js]: Removed thanks @Amzani

**Comment by Amzani** [comment]
Let's wait for other maintainers review as well @Shurtu-gal @Souvikns

**Review by Souvikns — APPROVED** [review]
LGTM 👍🏼

**Review by Shurtu-gal — APPROVED** [review]
LGTM

**Comment by AayushSaini101** [comment]
@Amzani Please review

**Review by Amzani — APPROVED** [review]
LGTM

**Comment by aeworxet** [comment]
@asyncapi/bounty_team

**Pull request merged**